src/org/zaproxy/zap/extension/alertFilters/AlertFilterAPI.java

```diff
@@ -0,0 +1,168 @@
+/*
+ * Zed Attack Proxy (ZAP) and its related class files.
+ * 
+ * ZAP is an HTTP/HTTPS proxy for assessing web application security.
+ * 
+ * Copyright 2013 The ZAP Development Team
+ * 
+ * Licensed under the Apache License, Version 2.0 (the "License"); 
+ * you may not use this file except in compliance with the License. 
+ * You may obtain a copy of the License at 
+ * 
+ *   http://www.apache.org/licenses/LICENSE-2.0 
+ *   
+ * Unless required by applicable law or agreed to in writing, software 
+ * distributed under the License is distributed on an "AS IS" BASIS, 
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. 
+ * See the License for the specific language governing permissions and 
+ * limitations under the License. 
+ */
+package org.zaproxy.zap.extension.alertFilters;
+
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+import net.sf.json.JSONObject;
+
+import org.apache.log4j.Logger;
+import org.zaproxy.zap.extension.api.ApiAction;
+import org.zaproxy.zap.extension.api.ApiException;
+import org.zaproxy.zap.extension.api.ApiException.Type;
+import org.zaproxy.zap.extension.api.ApiImplementor;
+import org.zaproxy.zap.extension.api.ApiResponse;
+import org.zaproxy.zap.extension.api.ApiResponseElement;
+import org.zaproxy.zap.extension.api.ApiResponseList;
+import org.zaproxy.zap.extension.api.ApiResponseSet;
+import org.zaproxy.zap.extension.api.ApiView;
+import org.zaproxy.zap.model.Context;
+import org.zaproxy.zap.users.User;
+import org.zaproxy.zap.utils.ApiUtils;
+
+/**
+ * The API for manipulating {@link User Users}.
+ */
+public class AlertFilterAPI extends ApiImplementor {
+
+	private static final Logger log = Logger.getLogger(AlertFilterAPI.class);
+
+	private static final String PREFIX = "alertFilter";
+
+	private static final String VIEW_ALERT_FILTER_LIST = "alertFilterList";
+
+	private static final String ACTION_ADD_ALERT_FILTER = "addAlertFilter";
+	private static final String ACTION_REMOVE_ALERT_FILTER = "removeAlertFilter";
+
+	private static final String PARAM_CONTEXT_ID = "contextId";
+	private static final String PARAM_RULE_ID = "ruleId";
+	private static final String PARAM_NEW_LEVEL = "newLevel";
+	private static final String PARAM_URL = "url";
+	private static final String PARAM_URL_IS_REGEX = "urlIsRegex";
+	private static final String PARAM_PARAMETER = "parameter";
+	private static final String PARAM_ENABLED = "enabled";
+
+	private ExtensionAlertFilters extension;
+
+	public AlertFilterAPI(ExtensionAlertFilters extension) {
+		super();
+		this.extension = extension;
+
+		this.addApiView(new ApiView(VIEW_ALERT_FILTER_LIST, null, new String[] { PARAM_CONTEXT_ID }));
```

Comment:
From the implementation of the view the parameter "contextId" should be mandatory.